### PR TITLE
feat(demo): Make demo more robust

### PR DESCRIPTION
**Pull request by Twixes**
## Changes

Makes the demo environment ready for the spotlight.

### CODE REVIEW

File: ee/clickhouse/models/group.py [hunk 1]
```diff
@@ -17,7 +17,9 @@ def create_group(
     group_key: str,
     properties: Optional[Dict] = {},
     timestamp: Optional[datetime.datetime] = None,
-) -> Group:
+    *,
+    clickhouse_only: bool = False,
```
Comment (tiina303) [inline ee/clickhouse/models/group.py:21]: instead let's create a function to create the group for CH and call that in here and wherever you need.

File: posthog/demo/matrix/manager.py [hunk 2]
```diff
@@ -58,54 +44,162 @@ def ensure_account_and_run(
...
-        set_time = time.time()  # FIXME
-        team.save()
-        print(f"[DEMO] Setting project up in {time.time() -set_time:.2f} s")
-        return team
+            self.save_sim_person(target_team, sim_person)
+
+    @classmethod
+    def prepare_master_team(cls):
+        if not Team.objects.filter(id=cls.MASTER_TEAM_ID).exists():
+            organization = Organization.objects.create(name="PostHog")
```
Comment (tiina303) [inline posthog/demo/matrix/manager.py:106]: sure we want to call it PostHog?

**Review by tiina303 — APPROVED** [review]
Didn't review super thoroughly, if you would like that then please smaller PRs :)

File: posthog/demo/hedgebox.py [hunk 3]
```diff
@@ -172,27 +196,23 @@ def _receive_file_link(self):
...
     MAX_RADIUS: int = 6
 
-    company_name: str
+    # None means the cluster is a social circle instead of a company
+    company_name: Optional[str]
 
     def __init__(self, *args, **kwargs):
         super().__init__(*args, **kwargs)
-        self.company_name = self.finance_provider.company()
+        is_company = self.random.random() < COMPANY_CLUSTERS_PROPORTION
```
[the remark this answers is not in the document]
Comment (Twixes) [reply posthog/demo/hedgebox.py:211]: That would be possible but would also require selecting the class at random, which would be some extra complexity to the overall structure… so I'll keep as is for now

File: posthog/demo/matrix/manager.py (hunk 2 above)
Comment (Twixes) [reply posthog/demo/matrix/manager.py:106]: I think it makes sense, especially if we make plugins managed in the demo environment like on Cloud, in which case they will this way say "Managed by PostHog"

File: ee/clickhouse/sql/person.py [hunk 4]
```diff
@@ -246,6 +246,34 @@
     f"INSERT INTO {PERSON_STATIC_COHORT_TABLE} (id, person_id, cohort_id, team_id, _timestamp) VALUES"
 )
 
+#
+# Copying demo data
+#
+
+COPY_PERSONS_BETWEEN_TEAMS = """
+INSERT INTO {table_name} (team_id, {columns_except_team_id}) SELECT %(target_team_id)s, {columns_except_team_id}
+FROM {table_name} WHERE team_id = %(source_team_id)s
```
[the remark this answers is not in the document]
Comment (Twixes) [reply ee/clickhouse/sql/person.py]: Great point, done!

File: ee/clickhouse/models/group.py (hunk 1 above)
Comment (Twixes) [reply ee/clickhouse/models/group.py:21]: Makes more sense, thanks

**Pull request merged**